### PR TITLE
SOV-3564 implement dllrTransferWithPermit in myntLib

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

**Review by night-pm — APPROVED** [review]
Approved

File: contracts/Dependencies/Mynt/IMassetManager.sol [hunk 1]
```diff
@@ -31,4 +31,6 @@ interface IMassetManager {
         uint256 _massetQuantity,
         address _recipient
     ) external returns (uint256 massetRedeemed);
+
+    function mAssetTokenTransferWithPermit() external view returns(address);
```
Comment (tjcloa) [inline contracts/Dependencies/Mynt/IMassetManager.sol]: it should be a getter from MassetManager

File: contracts/Dependencies/Mynt/MyntLib.sol [hunk 2]
```diff
@@ -29,10 +29,14 @@ library MyntLib {
         address _toToken,
```
Comment (tjcloa) [inline contracts/Dependencies/Mynt/MyntLib.sol:29]: fix typo above: 
`@param _toToken bAsset token address to wothdraw from DLLR` -> `@param _toToken bAsset token address to withdraw from DLLR`

File: contracts/Dependencies/Mynt/MyntLib.sol [hunk 3]
```diff
@@ -29,10 +29,14 @@ library MyntLib {
...
         IMassetManager.PermitParams calldata _permitParams
     ) internal returns (uint256) {
+        IDLLR dllrTransferWithPermit = IDLLR(_myntMassetManager.mAssetTokenTransferWithPermit());
         IDLLR dllr = IDLLR(_myntMassetManager.getToken());
+
         uint256 thisBalanceBefore = dllr.balanceOf(address(this));
         address thisAddress = address(this);
-        dllr.transferWithPermit(
+
+        IDLLR tokenTransferWithPermit = address(dllrTransferWithPermit) != address(0) ? dllrTransferWithPermit : dllr;
```
Comment (tjcloa) [inline contracts/Dependencies/Mynt/MyntLib.sol]: nice solution, but considering that FE needs to know which contract to ask permission for in advance, and querying the address on each call is an overkill, pls consider the following redesign:
1. save original function with name `redeemZusdDirectlyFromDllrWithPermit`, but reverting with the message - "use redeemZusdFromDllrWithPermit function" and comment that this revert can be removed, the function redeemZusdFromDllrWithPermit removed and `redeemZusdDirectlyFromDllrWithPermit` renamed to `redeemZusdFromDllrWithPermit` only if the aggregator token (DLLR) has no griefing attack vulnerablility and a link to the commit with this fix for clarity on how to check it
2. replace dllr with dllrTransferWithPermit and add a comment that Permit should be issued for the contract dllrTransferWithPermit

File: contracts/TestContracts/MassetManagerTester.sol [hunk 4]
```diff
@@ -76,4 +122,8 @@ contract MassetManagerTester is IMassetManager {
 
         return _massetQuantity;
     }
+
+    function mAssetTokenTransferWithPermit() external view override returns(address) {
```
Comment (tjcloa) [inline contracts/TestContracts/MassetManagerTester.sol]: should be a getter - see Masset review comments

**Review by tjcloa — CHANGES_REQUESTED** [review]
almost there, just a few comments

File: contracts/Dependencies/Mynt/IMassetManager.sol (hunk 1 above)
Comment (cwsnt) [reply contracts/Dependencies/Mynt/IMassetManager.sol]: updated in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/1/commits/12bbb6a0c7f6cef38929c6e2ce63905060be778e)

File: contracts/Dependencies/Mynt/MyntLib.sol (hunk 2 above)
Comment (cwsnt) [reply contracts/Dependencies/Mynt/MyntLib.sol:29]: updated in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/1/commits/12bbb6a0c7f6cef38929c6e2ce63905060be778e)

File: contracts/TestContracts/MassetManagerTester.sol (hunk 4 above)
Comment (cwsnt) [reply contracts/TestContracts/MassetManagerTester.sol]: updated in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/1/commits/12bbb6a0c7f6cef38929c6e2ce63905060be778e)

File: contracts/Dependencies/Mynt/MyntLib.sol (hunk 3 above)
Comment (cwsnt) [reply contracts/Dependencies/Mynt/MyntLib.sol]: From FE side, the behaviour would be the same, Except the fact that they just need to use DllTransferWithPermit as sender when generating signature

**Review by tjcloa — APPROVED** [review]

File: tests/js/StabilityPoolTest.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ const timeMachine = require("ganache-time-traveler");
 
 const MassetManagerTester = artifacts.require("MassetManagerTester");
 const NueMockToken = artifacts.require("NueMockToken");
+const NueMockTokenTransferWithPermit = artifacts.require("NueMockTokenTransferWithPermit");
```
Comment (tjcloa) [inline tests/js/StabilityPoolTest.js]: not used

File: tests/js/TroveManagerTest.js [hunk 6]
```diff
@@ -49,6 +50,7 @@ contract('TroveManager', async accounts => {
   let hintHelpers;
   let massetManager;
   let nueMockToken;
+  let nueMockTokenTansferWithPermit;
```
Comment (tjcloa) [inline tests/js/TroveManagerTest.js]: not used

**Review by tjcloa — APPROVED** [review]
lgtm

File: tests/js/TroveManagerTest.js (hunk 6 above)
Comment (cwsnt) [reply tests/js/TroveManagerTest.js]: removed

File: tests/js/StabilityPoolTest.js (hunk 5 above)
Comment (cwsnt) [reply tests/js/StabilityPoolTest.js]: removed

**Review by dharkmattr — APPROVED** [review]

**Pull request closed without merging**